controllers/machinedeployment_sync.go

```diff
@@ -143,12 +144,26 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(ctx context.Context, d *c
 		return nil, err
 	}
 	machineTemplateSpecHash := fmt.Sprintf("%d", hash)
+
+	// If reconciliationTime is after rolloutAfter then a rollout should happen or has already happened.
+	// To trigger it we include the rolloutAfter hash into the MachineSet name so the first time that
+	// reconciliationTime is after rolloutAfter then a new MachineSet is created and the rollout is orchestrated as usual.
```

Comment:
I'm sure I'm missing something. How does adding the rolloutAfterHash to the name make sure a new rollout is triggered?

Or is it rather that the changes in `FindNewMachineSetWithRolloutAfter` will "trigger" the rollout and we need a new name because otherwise the `Create` call in l.214 would fail because we already have a MS with the same name?

Reply:
Yup the latter is right. Updated the comment, let me know if that's still not clear.